estimagic/differentiation/numdiff_np.py

```diff
@@ -0,0 +1,240 @@
+    evals = np.transpose(evals, axes=(0, 1, 3, 2))
+    evals = namedtuple_from_kwargs(pos=evals[0], neg=evals[1])
+
+    jac_candidates = {}
+    for m in ["forward", "backward", "central"]:
+        jac_candidates[m] = getattr(fd, f"jacobian_{m}")(evals, steps, f0)
+
+    orders = {
+        "central": ["central", "forward", "backward"],
+        "forward": ["forward", "backward"],
+        "backward": ["backward", "forward"],
+    }
+
+    if n_steps == 1:
+        jac = _consolidate_one_step_derivatives(jac_candidates, orders[method])
+    else:
+        raise NotImplementedError("Extrapolation is not yet implemented.")
+
+    res = jac.flatten() if f_was_scalar else jac
+
+    return res
+
+
+def _consolidate_one_step_derivatives(candidates, preference_order):
+    """Replace missing derivative estimates of preferred method with others.
+
+    Args:
+        candidates (dict): Dictionary with derivative estimates from different methods.
+        preference_order (list): Order on (a subset of) the keys in candidates. Earlier
+            entries are preferred.
+
+    Returns:
+        consolidated (np.ndarray): Array of same shape as input derivative estimates.
+
+    """
+    preferred, others = preference_order[0], preference_order[1:]
+    consolidated = candidates[preferred].copy()
+    for other in others:
+        consolidated = np.where(
+            np.isfinite(consolidated), consolidated, candidates[other]
+        )
+
+    return consolidated.reshape(consolidated.shape[1:])
+
+
+def _consolidate_extrapolated(candidates):
+    """Get the best possible derivative estimate, given an error estimate.
+
+    See https://tinyurl.com/ubn3nv5 for corresponding code in numdifftools and
+    https://tinyurl.com/snle7mb for an explanation of how errors of Richardson
+    extrapolated derivative estimates can be estimated.
+
+    """
+    raise NotImplementedError
+
+
+def _nan_skipping_batch_evaluator(func, arglist, n_processes):
+    """Evaluate func at each entry in arglist, skipping np.nan entries.
+
+    The function is only evaluated at inputs that are not a scalar np.nan.
+    The outputs corresponding to skipped inputs as well as for inputs on which func
+    returns a scalar np.nan are arrays of the same shape as the result of func, filled
+    with np.nan.
+
+    Args:
+        func (function): Python function that returns a numpy array. The shape
+            of the output of func has to be the same for all elements in arglist.
+        arglist (list): List with inputs for func.
+        n_processes (int): Number of processes.
+
+    Returns
+        evaluations (list): The function evaluations, same length as arglist.
+
+    """
+    # extract information
+    nan_indices = {
+        i for i, arg in enumerate(arglist) if isinstance(arg, float) and np.isnan(arg)
+    }
+    real_args = [arg for i, arg in enumerate(arglist) if i not in nan_indices]
+
+    # evaluate function
+    if n_processes == 1:
```

Comment:
No need for the case differentiation. Parallel becomes a for-loop for `n_jobs=1`.